crates/worker/src/partition/state_machine/command_interpreter/mod.rs

```diff
@@ -755,6 +862,39 @@ where
                 // where the invocation should be executed
                 self.handle_invoke(effects, state, service_invocation).await
             }
+            Timer::CleanInvocationStatus(invocation_id) => {
```

Comment:
If the `CleanInvocationStatus` would also have a `Option<IdempotencyId>` field, then this case could look like:

```
effects.free_invocation(invocation_id);
effects.delete_idempotency_id(idempotency_id);
```

which saves a read from the state storage.

Reply:
I prefer to go down this road for the time being, as i wanna revisit the whole id business with https://github.com/restatedev/restate/issues/1329. We can optimize this later if we want to.